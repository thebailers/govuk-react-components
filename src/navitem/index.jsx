import React from 'react';
import PropTypes from 'prop-types';

export const navItemOf = (Type) => {
  const component = (props) => {
    const {active, className = '', ...other} = props;
    const elemClass = (active ? 'active' : '') + (className ? ' ' + className : '');
    return (
      <li>
        <Type {...other} className={elemClass}>{props.children}</Type>
      </li>
    )
  };
  component.defaultName = 'NavItemOf' + Type;
  component.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    active: PropTypes.bool
  };
  return component;
};

export const NavItem = navItemOf('a');
NavItem.propTypes = Object.assign({ href: PropTypes.string.isRequired }, NavItem.propTypes);

export default NavItem;
