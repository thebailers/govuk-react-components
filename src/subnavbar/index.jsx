import React from 'react';
import PropTypes from 'prop-types';
import NavItemA from '../navitem';

export const SubNavBar = (props) => (
  <div className={`navbar${props.className ? ' ' + props.className : ''}`}>
    <ul className="navbar__list-items">{props.children}</ul>
  </div>
);

SubNavBar.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default SubNavBar;
export const NavItem = NavItemA;
