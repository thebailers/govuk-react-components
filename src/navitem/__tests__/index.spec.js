import React from 'react'
import {render} from 'enzyme'
import {SubNavBar} from 'govuk-react-components/subnavbar';
import {NavItem} from 'govuk-react-components/navitem';

describe('The NavItem Component', () => {
  it('should render an included classname', () => {
    expect(
      render(
        <SubNavBar>
          <NavItem href="#test" className="test class-name">Item A</NavItem>
        </SubNavBar>
      )
    ).toMatchSnapshot()
  });

  it('should render with an active class if the active flag is true', () => {
    expect(
      render(
        <SubNavBar>
          <NavItem href="#test" className="test class-name" active>Item A</NavItem>
        </SubNavBar>
      )
    ).toMatchSnapshot()
  });
});

