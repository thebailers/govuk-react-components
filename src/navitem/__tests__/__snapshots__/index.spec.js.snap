// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NavItem Component should render an included classname 1`] = `
<div
  class="navbar"
>
  <ul
    class="navbar__list-items"
  >
    <li>
      <a
        class=" test class-name"
        href="#test"
      >
        Item A
      </a>
    </li>
  </ul>
</div>
`;

exports[`The NavItem Component should render with an active class if the active flag is true 1`] = `
<div
  class="navbar"
>
  <ul
    class="navbar__list-items"
  >
    <li>
      <a
        class="active test class-name"
        href="#test"
      >
        Item A
      </a>
    </li>
  </ul>
</div>
`;
